RealmSwift/Impl/SchemaDiscovery.swift

```diff
@@ -60,6 +60,28 @@ internal extension RLMProperty {
             self.updateAccessors()
         }
     }
+
+    /// Exposed for Macros.
+    /// Important: Keep args in same order & default value as `@Persisted` property wrapper
+    public convenience init<O: ObjectBase, V: _Persistable>(
```

Comment:
I have not actually verified that this works, but if it does this should be `@_spi(RealmSwiftPrivate)`  so that code using it has to do `@_spi(RealmSwiftPrivate) import RealmSwift` to reflect that this isn't a stable part of the API.

Reply:
Love this. In fact, I've pushed another commit to remove `_customRealmProperties` from the Objective-C files and done it as an extension in RealmSwift and annotated it with this `@_spi`. Since we only want to expose it to Swift Macros, makes sense yeah? 

What do you think?